adam/perception/high_level_semantics_situation_to_developmental_primitive_perception.py

```diff
@@ -278,6 +282,49 @@ def _get_relations_from_condition(
 
         return relations
 
+    def _find_perception_object_for_situation_object(
+        self, slot: SituationObject, entities_to_roles, action_roles_to_fillers
+    ) -> ObjectPerception:
+        if slot in entities_to_roles.keys():
+            filler_role_in_action = only(entities_to_roles[slot])
+            situation_object = only(action_roles_to_fillers[filler_role_in_action])
+            perception = self._objects_to_perceptions[situation_object]
+        else:
+            found_objects = []
```

Comment:
perhaps `objects_matching_constraints`.
Also add a comment above the `if` that if we already know what the situation object is, we just look it up. Otherwise, we need to search through the other situation objects to see if any matches the constraints we know to hold for the situation object.
Note also we don't currently handle constraints due to relations and link to the relevant issue